protocol/messenger.go

```diff
@@ -492,7 +493,15 @@ func NewMessenger(
 		encryptor: encryptionProtocol,
 	}
 
-	communitiesManager, err := communities.NewManager(identity, installationID, database, encryptionProtocol, logger, ensVerifier, c.communityTokensService, transp, transp, communitiesKeyDistributor, c.torrentConfig, managerOptions...)
+	communitiesManager, err := communities.NewManager(identity, installationID, database, encryptionProtocol, logger, ensVerifier, c.communityTokensService, transp, transp, communitiesKeyDistributor, managerOptions...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Depending on the OS go will choose whether to use the "communities/manager_torrent_mobile.go" or
+	// "communities/manager_torrent.go" version of this function based on the build instructions for those files.
+	// See those file for more details.
+	torrentManager, err := communities.NewTorrentManager(c.torrentConfig, logger, communitiesManager.GetPersistence(), transp, identity, encryptionProtocol, communitiesManager)
```

Comment:
I thought about this also. The main reason why I decided to separate `TorrentManager` entirely was because after I'd separated everything into `TorrentManager` from `communities .Manager` both structs seems to perform very different functionality. I considered if `TorrentManager` was exclusively responsible for archiving community history and because `communities .Manager` didn't call a single "torrent" function I decided that `Messenger` seemed be the place where calling the torrent manage was naturally happening.

Example see below:

```go
func (m *Messenger) EditCommunity(request *requests.EditCommunity) (*MessengerResponse, error) {
	if err := request.Validate(); err != nil {
		return nil, err
	}

	community, err := m.communitiesManager.EditCommunity(request)
	if err != nil {
		return nil, err
	}

	communitySettings := communities.CommunitySettings{
		CommunityID:                  community.IDString(),
		HistoryArchiveSupportEnabled: request.HistoryArchiveSupportEnabled,
	}
	err = m.communitiesManager.UpdateCommunitySettings(communitySettings)
	if err != nil {
		return nil, err
	}

	id := community.ID()

	if m.torrentManager.IsReady() {
		if !communitySettings.HistoryArchiveSupportEnabled {
			m.torrentManager.StopHistoryArchiveTasksInterval(id)
		} else if !m.torrentManager.IsSeedingHistoryArchiveTorrent(id) {
			var communities []*communities.Community
			communities = append(communities, community)
			go m.InitHistoryArchiveTasks(communities)
		}
	}

	response := &MessengerResponse{}
	response.AddCommunity(community)
	response.AddCommunitySettings(&communitySettings)
	err = m.SyncCommunitySettings(context.Background(), &communitySettings)
	if err != nil {
		return nil, err
	}

	return response, nil
}
```

In this case ideally all "torrent" calls should be pushed into the respective `communities .Manager` function calls and not triggered by `Messenger`.

```go
if m.torrentManager.IsReady() {
	if !communitySettings.HistoryArchiveSupportEnabled {
		m.torrentManager.StopHistoryArchiveTasksInterval(id)
	} else if !m.torrentManager.IsSeedingHistoryArchiveTorrent(id) {
		var communities []*communities.Community
		communities = append(communities, community)
		go m.InitHistoryArchiveTasks(communities)
	}
}
```

In this way `Messenger` doesn't need to know that there is an archive. But this would require considerable additional refactoring. This could be the goal of a new round of refactoring, I've already scoped the places where torrent functionality is consumed / called directly from `Messenger`

See here https://github.com/status-im/status-go/pull/5257#issuecomment-2143956872

What do you think @igor-sirotin ?

Reply:
We don't have to refactor further in this PR of course